### ISSUE TITLE
Upgrade transitive connectivity_plus dependency to 5.0.x

### ISSUE THREAD

**Issue by DenizKucukozturk** [GetStream/stream-chat-flutter#1752]
### Which packages are you using?

stream_chat_flutter

### On what platforms did you experience the issue?

iOS

### What version are you using?

stream_chat_flutter: ^6.10.0

### What happened?

`connectivity_plus` 4.x.x version has a known problem with Xcode15. It can be seen in this issue https://github.com/fluttercommunity/plus_plugins/issues/2162. They already fixed it and upgraded the package but because of this transitive dependency I still can't build my project on Xcode15 with a real iOS device.

### Steps to reproduce

```bash
Build and run any project with stream_chat_flutter: ^6.10.0 on iOS with Xcode 15
```

### Relevant log output

```shell
[debug] /tmp/jenkins/IRIS-SDK/base/internal/log/iris_logger.cc:35 InitLogger /var/mobile/Containers/Data/Application/6EAD1EB4-4605-4006-BC08-9165BD53A1B2/Library/Caches/
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x0000000000000000
error: memory read failed for 0x0
Target 0: (Runner) stopped.
```

### Flutter doctor output

```shell
[✓] Flutter (Channel stable, 3.13.6, on macOS 13.5.2 22G91 darwin-arm64, locale en-TR)
    • Flutter version 3.13.6 on channel stable at /Users/denizkucukozturk/fvm/versions/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ead455963c (3 weeks ago), 2023-09-26 18:28:17 -0700
    • Engine revision a794cf2681
    • Dart version 3.1.3
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/denizkucukozturk/Library/Android/sdk
    • Platform android-34, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.79.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • sdk gphone64 arm64 (mobile) • emulator-5556 • android-arm64  • Android 13 (API 33) (emulator)
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 118.0.5993.70

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by deep-gaurav** [comment]
This is a blocker for us too.
There is a dependabot PR to fix this open #1750

**Comment by esarbanis** [comment]
#1750 just landed in `develop` branch
